src/main/java/reposense/parser/CliArguments.java

```diff
@@ -0,0 +1,139 @@
+package reposense.parser;
+
+import java.io.File;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Optional;
+
+import reposense.exception.ParseException;
+
+/**
+ * Represents command line arguments user supplied when running the program.
+ */
+public class CliArguments {
+
+    private static final String ARGUMENT_PREFIX_DASH = "-";
+
+    private static final String CONFIG_FILE_ARG = "config";
+    private static final String TARGET_FILE_ARG = "output";
+    private static final String SINCE_DATE_ARG = "since";
+    private static final String UNTIL_DATE_ARG = "until";
+
+    private static final String PARSE_EXCEPTION_MESSAGE_INVALID_INPUTS = "Failed to parse inputs arguments";
+    private static final String PARSE_EXCEPTION_MESSAGE_NO_CSV_FILE = "Failed due to missing CSV file";
+
+    private static final String DEFAULT_FILE_ARG = ".";
+
+    private File configFile;
+    private File targetFile;
+    private Optional<Date> sinceDate;
+    private Optional<Date> untilDate;
+
+    /**
+     * Parses user-supplied arguments.
+     *
+     * @throws IllegalArgumentException if the given args inputs are either
+     *  - malformed
+     *  - fail to parse
+     *  - missing mandatory fields
+     */
+    public CliArguments(String[] args) throws IllegalArgumentException {
+        sinceDate = Optional.empty();
+        untilDate = Optional.empty();
+        targetFile = new File(DEFAULT_FILE_ARG);
+
+        final HashMap<String, String> argumentMap = generateArgumentMap(args);
+        checkAllMandatoryArgumentsPresent(argumentMap);
+        setUserInputValuesToArgument(argumentMap);
+    }
+
+    /**
+     * Verifies if {@code args} begins with a dash character and follows by an input and generates an ArgumentMap
+     *
+     * @throws IllegalArgumentException If the given args are malformed,
+     * which means it does not have a dash character or does not follow by an input.
+     */
+    private HashMap<String, String> generateArgumentMap(String[] args) {
```

Comment:
Would prefer this method to be in a separate parser class but I guess we can do this afterwards.